libcloud/compute/drivers/gridscale.py

```diff
@@ -0,0 +1,977 @@
+        :rtype: ``bool``
+        """
+        result = self._sync_request(endpoint='objects/servers/{}'
+                                    .format(node.id),
+                                    method='DELETE')
+        return result.status == 204
+
+    def destroy_volume(self, volume):
+        """
+        Delete volume.
+
+        :param volume: Volume to be destroyed.
+        :type volume: :class:`.StorageVolume`
+
+        :return: True if the destroy was successful, otherwise False.
+        :rtype: ``bool``
+        """
+        result = self._sync_request(endpoint='objects/storages/{}'
+                                    .format(volume.id),
+                                    method='DELETE')
+        return result.status == 204
+
+    def ex_destroy_ip(self, ip):
+        """
+        Delete an ip.
+
+        :param ip: IP object.
+        :type ip: :class:`.GridscaleIp`
+
+        :return: ``True`` if delete_image was successful, ``False`` otherwise.
+        :rtype: ``bool``
+        """
+        result = self._sync_request(endpoint='objects/ips/{}'
+                                    .format(ip.id),
+                                    method='DELETE')
+        return result.status == 204
+
+    def destroy_volume_snapshot(self, snapshot):
+        """
+        Destroy a snapshot.
+
+        :param snapshot: The snapshot to delete.
+        :type snapshot: :class:'.VolumeSnapshot`
+
+        :return: True if the destroy was successful, otherwise False.
+        :rtype: ``bool``
+        """
+        result = self._sync_request(endpoint='objects/storages/'
+                                             '{}/snapshots/{}/'
+                                    .format(snapshot.extra['parent_uuid'],
+                                            snapshot.id),
+                                    method='DELETE')
+        return result.status == 204
+
+    def ex_destroy_networks(self, network):
+        """
+        Delete network.
+
+        :param network: Network object.
+        :type network: :class:`.GridscaleNetwork`
+
+        :return: ``True`` if destroyed successfully, otherwise ``False``
+        :rtype: ``bool``
+        """
+        result = self._sync_request(endpoint='objects/networks/{}'
+                                    .format(network.id),
+                                    method='DELETE')
+        return result.status == 204
+
+    def delete_image(self, node_image):
+        """
+        Destroy an image.
+
+        :param node_image: Node image object.
+        :type node_image: :class:`.NodeImage`
+
+        :return: True if the destroy was successful, otherwise False
+        :rtype: ``bool``
+
+        """
+        result = self._sync_request(endpoint='objects/templates/{}'
+                                    .format(node_image.id),
+                                    method='DELETE')
+        return result.status == 204
+
+    def patch_node(self, node, name=None):
```

Comment:
This method and ``patch_storage`` is not part of the standard Libcloud API, so it should be prefixed with ``ex_`` (``ex_patch_node``, ``ex_patch_storage``).

Actually, it looks like all those two methods do is rename a node / storage volume, so they should probably be called something along the lines of ``ex_rename_node`` and ``ex_rename_volume``.